libevmasm/RuleList.h

```diff
@@ -374,6 +390,32 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart7(
 		false
 	});
 
+	rules.push_back({
+		// AND(A, SHR(B, X)) -> SHR(B, X) if 2^(256-B)-1 == A
```

Comment:
Still the wrong comment ;-). (The comment is correct, but the rule now catches more cases)
I think that's simplest:
``A & ((2^256-1) >> B) == ((2^256-1) >> B)``
It's readable enough, it's correct and it makes it easier to confirm the condition below anyways.